src/main_thread/api/debug/buffer_graph.ts

```diff
@@ -150,13 +150,14 @@ export default class SegmentSinkGraph {
   }
 
   private _getColorForRepresentation(representation: IRepresentation): string {
-    const color = this._colorMap.get(representation);
+    const serializableRepresentation = JSON.stringify(representation);
+    const color = this._colorMap.get(serializableRepresentation);
```

Comment:
because `representation` is coming from the worker, it has been serialized, and parsed to a javascript object.
If it's used as a key in a weakmap, it will not be able to recognize the same representation because due to the serializing/parsing there is now 2 different objects (with the same shape and values).

Solution was to perform check for "deep equality" by stringifying the object

Reply:
Isn't  `representation.uniqueId` sufficient to identify a given Representation?

Reply:
Not that much of a fan of deep equality checking by JSON serialization as the order of properties might be different despite Representation being equal.

It also seems too "indirect" to me for equality checking, I would be afraid it may break unexpectedly as it's not the role of `JSON.stringify`.

Reply:
Yes definitly the good solution is to use `uniqueId`